VSDiagnostics/VSDiagnostics/VSDiagnostics/Utilities/Extensions.cs

```diff
@@ -154,21 +186,26 @@ public static bool IsDefinedInAncestor(this IMethodSymbol methodSymbol)
             foreach (var @interface in interfaces)
             {
                 var interfaceMethods =
-                    @interface.GetMembers().Select(containingType.FindImplementationForInterfaceMember).Where(x => x != null);
+                    @interface.GetMembers().Select(containingType.FindImplementationForInterfaceMember).ToList();
 
-                if (interfaceMethods.Any(method => method.Equals(methodSymbol)))
+                foreach (var interfaceMethod in interfaceMethods)
                 {
-                    return true;
+                    if (interfaceMethod != null && interfaceMethod.Equals(methodSymbol))
+                    {
+                        return true;
+                    }
                 }
             }
 
             var baseType = containingType.BaseType;
             while (baseType != null)
             {
-                var baseMethods = baseType.GetMembers().OfType<IMethodSymbol>();
-                if (baseMethods.Any(method => method.Equals(methodSymbol.OverriddenMethod)))
+                foreach (var baseMethod in baseType.GetMembers().OfType<IMethodSymbol>())
```

Comment:
`IMethodSymbol` is used for any "method-like symbol", including ctors and properties.  My custom `OfType` works only for known SyntaxKind types.